ko/sql/tuning.rst

```diff
@@ -3831,6 +3831,160 @@ N:1 관계의 **LEFT OUTER JOIN**\에서 조인 조건 외에 오른쪽 테이
+
+
+따라서 쿼리 1의 뷰 쿼리 블록은 뷰를 참조하는 쿼리 블록과의 머지(merge) 과정을 거쳐
+쿼리 2와 같은 형태로 변환된다. 
+
+이를 **View Merging**\이라고 한다. 이 과정을 거치면 옵티마이저는 더 많은 액세스 경로를 조사 대상으로 삼을 수 있다.
+
+
+현재 큐브리드에서는 쿼리가 다음 조건에 해당하면 **View Merging**\을 수행할 수 없다.
+
+
+    #. **CONNECT BY**\를 포함한 경우
+
+    #. 인라인 뷰가 **DISTINCT**\ 문을 포함한 경우
+
+    #. **CTE**\(Common Table Expressions)가 쿼리에 포함되어 있는 경우
+    
+    #. 뷰를 **OUTER JOIN**\ 하는 경우
+
+    #. 집계함수나 분석함수를 사용하는 경우
+
+    #. **ROWNUM, LIMIT**\ 또는 **GROUPBY_NUM ()**\ 을 사용하는 경우
+
+    #. **correlated subquery**\ 를 사용하여 작성된 경우
+
+    #. 인라인 뷰에 **FROM**\ 절이 없는 경우
+
+    #. 인라인 뷰가 메소드를 포함한 경우
+
+다음은 **CONNECY BY**\를 포함한 예시이다.
+
+.. code-block:: sql
+
+    -- Creating tree table and then inserting data
+    CREATE TABLE tree(ID INT, MgrID INT, Name VARCHAR(32), BirthYear INT);
+
+    INSERT INTO tree VALUES (1,NULL,'Kim', 1963);
+    INSERT INTO tree VALUES (2,NULL,'Moy', 1958);
+    INSERT INTO tree VALUES (3,1,'Jonas', 1976);
+    INSERT INTO tree VALUES (4,1,'Smith', 1974);
+    INSERT INTO tree VALUES (5,2,'Verma', 1973);
+    INSERT INTO tree VALUES (6,2,'Foster', 1972);
+    INSERT INTO tree VALUES (7,6,'Brown', 1981);
+
+    -- Executing a hierarchical query with CONNECT BY clause
+    SELECT *
+    FROM (SELECT * FROM tree WHERE BirthYear = 1973) t
+    CONNECT BY PRIOR t.id=t.mgrid; 
+
+
+위 쿼리에서는 **CONNECY BY**\절을 사용하기 때문에 **View Merging**\을 수행할 수 없다.
+
+다음은 인라인 뷰가 **DISTINCT**\문을 포함한 예시이다.
+
+.. code-block:: sql
+
+    SELECT * FROM (SELECT DISTINCT BirthYear FROM Tree) T;
+
+위 쿼리의 인라인 뷰 내부에 **DISTINCT**\문이 사용되어 **View Merging**\을 수행할 수 없다.
+
+
+다음은 **CTE**\(Common Table Expressions)가 쿼리에 포함되어 있는 예시이다.
+
+
+.. code-block:: sql
+
+    WITH cte AS (SELECT * FROM emp WHERE job = 'SALESMAN')
+    SELECT * FROM cte WHERE cte.deptno = 10;
+
+위와 같이 **CTE**\를 포함한 쿼리는 **View Merging**\을 수행할 수 없다.
+
+다음은 뷰를 **OUTER JOIN**\ 하는 예시이다.
+
+.. code-block:: sql
+
+    SELECT * FROM emp a 
+    LEFT OUTER JOIN (SELECT * FROM dept WHERE loc = 'CHICAGO') b 
+    ON a.deptno = b.deptno;
+
+위와 같이 명시적으로 **OUTER JOIN**\을 수행하는 경우에는 **View Merging**\을 수행할 수 없다.
+
+다음은 집계함수나 분석함수를 사용하는 예시이다.
+
+.. code-block:: sql
+
+    SELECT *
+    FROM (SELECT AVG(salary) FROM emp WHERE job = 'SALESMAN') a;
+
+집계함수나 분석함수를 포함한 쿼리의 경우 **View Merging**\의 대상이 되지 않는다.
+
+다음은 **ROWNUM, LIMIT**\ 또는 **GROUPBY_NUM ()**\을 사용하는 예시이다.
+
+.. code-block:: sql
+
+    SELECT *
+    FROM (SELECT gender, rownum FROM athlete WHERE rownum < 15) a
+    WHERE gender = 'M';
+
+**ROWNUM, LIMIT**\ 또는 **GROUPBY_NUM ()**\을 사용한 쿼리의 경우 **View Merging**\이 불가능하다.
+
+다음은 **correlated subquery**\ 를 사용하여 작성된 예시이다
+
+.. code-block:: sql
+
+    SELECT * FROM emp a 
+    WHERE EXISTS (SELECT 1 FROM dept b WHERE a.deptno = b.deptno);
+
+**correlated subquery**\를 사용한 쿼리의 경우 **View Merging**\이 불가능하다.
+
+다음은 인라인 뷰에 **FROM**\ 절이 없는 예시이다.
+
+.. code-block:: sql
+
+    SELECT 1 + (SELECT 2);
+
+인라인 뷰에 **FROM**\절이 없는 경우 **View Merging** 하여 인덱스 스캔을 사용 가능하게 하는 등의 성능 최적화가 불가능하여 **View Merging**\을 수행하지 않는다.
+
```

Comment:
need to add random function case

Reply:
I added random, drandom, sys_guid function cases.